### DISCUSSION TITLE
App state set value not work on Modal after upgraded

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3633] [Help]
Hello,

Just upgraded from 0.9.173 to 1.0.2, and found the app state `set value` just didn't work while open as Modal, but works open as page(Full screen). 

I tried downgraded to 0.9.189, still the same. Then revert to 0.9.173 and finally works as before.

May I know if this is an expected change, or it's a bug?

**Comment by mjashanks** [comment]
Hi Sam. I have reproduced your issue: https://github.com/Budibase/budibase/issues/3639. The workaround is to use the "Persist this value" checkbox.

While investigating this, I discovered another bug: https://github.com/Budibase/budibase/issues/3636

This means that if you are changing your `Update State` action, you'll actually need to remove and re-add it. Updating the `Update State` action is not working.

**Reply by samg666** [reply]
@mjashanks  Michael, I am always use persistent app state between different pages, the value shared between pages without issue, but in some place it's open as modal, it also works on 0.9.173, just didn't work on 1.0.2. 

I have just downgraded back to 0.9.173, it works again. It should be a bug.
